--- wget-1.21.2/src/html-url.c	2021-09-07 21:56:08.000000000 +0300
+++ wget-1.21.2/src/html-url-extended.c	2021-11-10 00:02:23.694259075 +0300
@@ -158,6 +158,8 @@
   { TAG_IMG,            "href",         ATTR_INLINE },
   { TAG_IMG,            "lowsrc",       ATTR_INLINE },
   { TAG_IMG,            "src",          ATTR_INLINE },
+  { TAG_IMG,            "data-src",     ATTR_INLINE },
+  { TAG_IMG,            "data-href",    ATTR_INLINE },
   { TAG_INPUT,          "src",          ATTR_INLINE },
   { TAG_LAYER,          "src",          ATTR_INLINE | ATTR_HTML },
   { TAG_OBJECT,         "data",         ATTR_INLINE },
